File: core/src/main/java/com/mcreater/amclcore/model/oauth/XBLTokenRequestModel.java
package com.mcreater.amclcore.model.oauth;

import com.mcreater.amclcore.annotations.OAuthLoginModel;
import com.mcreater.amclcore.annotations.RequestModel;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Builder
@Data
@OAuthLoginModel
@RequestModel
public class XBLTokenRequestModel {
    private String IssueInstant;
    private String NotAfter;
    private String Token;
    private XBLTokenDisplayClaimsModel DisplayClaims;

    @Builder
    @Data
    @OAuthLoginModel
    @RequestModel
    public static class XBLTokenDisplayClaimsModel {
        private List<XBLTokenUserHashModel> xui;
    }

    @Builder
    @Data
    @OAuthLoginModel
    @RequestModel
    public static class XBLTokenUserHashModel {
        private String uhs;
    }
}

File: core/src/main/java/com/mcreater/amclcore/model/oauth/XBLTokenRequestModel.kt
package com.mcreater.amclcore.model.oauth

import com.mcreater.amclcore.annotations.OAuthLoginModel
import com.mcreater.amclcore.annotations.RequestModel

@OAuthLoginModel
@RequestModel
data class XBLTokenRequestModel(
    var IssueInstant: String? = null,
    var NotAfter: String? = null,
    var Token: String? = null,
    var DisplayClaims: XBLTokenDisplayClaimsModel? = null
) {
    @OAuthLoginModel
    @RequestModel
    data class XBLTokenDisplayClaimsModel(
        var xui: List<XBLTokenUserHashModel>? = null
    )

    @OAuthLoginModel
    @RequestModel
    data class XBLTokenUserHashModel(
        var uhs: String? = null
    )
}
